### PR TITLE
rename "master" to "main"

### PR DESCRIPTION
**Pull request by jvanasco**
See https://github.com/Pylons/pyramid/issues/3712

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ build-backend = "setuptools.build_meta"
 [tool.black]
 line-length = 79
 skip-string-normalization = true
-target-version = ['py27', 'py35', 'py36', 'py37', 'py38']
+target-version = ['py35', 'py36', 'py37', 'py38']
```
Comment (stevepiercy) [inline pyproject.toml:8]: Pyramid 1.10 still supports Python 2.7. I don't think we can drop it. Let's see what maintainers say.

**Comment by jvanasco** [comment]
The following will need maintainers to implement/address:

* The current version of black does not support py27, so we either version-pin black or drop py27
* Coverage had a regression a while back. see https://github.com/pytest-dev/pytest-cov/issues/222  I am not familiar enough with coverage to handle a fix
* There seems to be an issue with py3.4 running on github.  i don't know why.  multiple pylons project packages are breaking in the same setup.

**Comment by mmerickel** [comment]
Yeah we need to drop py27,py34,py35,py36, and add py39,py310,py311.

**Comment by jvanasco** [comment]
Redefining the supported versions may be better done in a separate PR, then followed by a rebase or similar of this PR.

The following things need to be handle for that change:

* /demo/demo.py defines "PY3" and has a docstring
  * # ease testing py2 and py3 in same directory
  * i think this may still be useful for serial test runs under a single platform, though I am not sure.
* the nosetests xml files will need to be generated for newer pythons. i don't know how to do that.
* /pyproject.toml needs the versions updated
* /setup.cfg needs the classifiers and python_requires updated; the install_requires might require pyramid2 
* /src/pyramid_debugtoolbar/compat.py should eventually go, but a first step would be dropping the PY3/PY2 blocks
* /src/pyramid_debugtoolbar/panels/request_vars.py has a docstring about PY2. might be okay to drop, but might be good to leave for anyone upgrading to PY3
* /src/pyramid_debugtoolbar/repr.py has `py2_string_repr` looks safe to drop; along with some blocks of PY3 from compat
* /tests/test_auth.py `test_unauthenticated` worked around a lack of "assertIsNone` in Py26; can be upgraded
* /tests/test_panels/test_request_vars.py - some PY3 blocks
* /tests/test_panels/test_session.py - some PY3 blocks
* /tests/test_panels/test_session.py `test_sorting_fatal` has a skipUnless for Py3. 
* /tests/test_panels/test_sqla.py - `Note: the <pre> styles are different on the Py2 and Py3 libraries`
* /tox.ini suggests this isn't run against pyramid2

File: pyproject.toml (hunk 1 above)
Comment (mmerickel) [reply pyproject.toml:8]: It's ok to drop it.

File: demo/test.py [hunk 2]
```diff
@@ -88,127 +91,140 @@ def tearDown(self):
...
+    # browser.open('/test_sqla')
+    # browser.wait_for_page_to_load("30000")
+    # self.visit_dbtb()
+    # browser.fire_event("css=a.pDebugSQLAlchemyPanel", 'click')
+    # browser.fire_event("link=EXPLAIN", 'click')
+    # browser.wait_for_condition(
+    # "selenium.isElementPresent(\"//table[@id='pSqlaTable']\")",
+    # "30000")
+    # result = browser.get_text('css=#pDebugWindow .pDebugPanelTitle h3')
+    # self.assertEqual(result, 'SQL Explained')
```
Comment (mmerickel) [inline demo/test.py:176]: This is a mess losing all of this indentation in the comments.

**Comment by mmerickel** [comment]
My current plan is to drop unsupported versions and fix CI before looking at this PR. Likely will end up closing this instead of merging it. Thank you for all of the notes in your previous comment which I'll double check as I go through that.

**Pull request closed without merging**